packages/common/src/common/toast.tsx

```diff
@@ -0,0 +1,217 @@
+import React from "react";
+import {
+  ToastContainer as TC,
+  toast as toastFn,
+  type Id,
+} from "react-toastify";
+import "react-toastify/dist/ReactToastify.css";
+import {
+  IconCheckCircleFill,
+  IconErrorFill,
+  IconInfoCircleFill,
+  Notification,
+  type NotificationType,
+} from "hds-react";
+import styled from "styled-components";
+
+type ToastProps = {
+  text: string;
+  type?: "success" | "error" | "alert" | "info";
+  label?: string;
+  ariaLabel?: string;
+  duration?: number;
+  dataTestId?: string;
+  options?: Record<string, unknown>;
+  icon?: JSX.Element;
+};
+
+const Container = styled(TC)`
+  [class^="Toastify__toast"] {
+    position: relative;
+    padding: 0;
+  }
+  --toastify-toast-bd-radius: 0;
+  --toastify-toast-width: max(30%, 320px);
+  --toastify-toast-min-height: 48px;
+  // We need to define the active color for the toast, to use it as the progress bar color...
+  --toastify-color-progress-info: rgb(0, 98, 185);
+  --toastify-color-progress-success: #007a64;
+  --toastify-color-progress-warning: #c2a251;
+  --toastify-color-progress-error: #bd2719;
+  // ...and use the same color for the icons
+  [class*="Toastify__toast--info"] {
+    color: var(--toastify-color-progress-info);
+  }
+  [class*="Toastify__toast--success"] {
+    color: var(--toastify-color-progress-success);
+  }
+  [class*="Toastify__toast--warning"] {
+    color: var(--toastify-color-progress-warning);
+  }
+  [class*="Toastify__toast--error"] {
+    color: var(--toastify-color-progress-error);
+  }
+  // Styles for the toastify icon to play nice with HDS style definitions
+  [class^="Toastify__toast-icon"] {
+    position: absolute;
+    width: var(--spacing-m);
+    height: var(--spacing-m);
+    top: calc(50% + 4px);
+    left: var(--spacing-s);
+    transform: translateY(-50%);
+    z-index: 1;
+    display: none;
```

Comment:
how does this work if it's always set to `display: none`? what changes it to something else?

Reply:
there's a `display: block;` a few rows after that, below the commented `&:not(:has...` selector.